### ISSUE TITLE
Proper icons for tsconfig.*.json (or: support wildcards in file name mappings)

### ISSUE THREAD

**Issue by Neonit** [jesseweed/seti-ui#556]
While `tsconfig.json` has a proper icon, similar files like `tsconfig.foo.json` don't, although this seems to be an accepted naming convention for alternative Typescript configuration files.

If I guessed correctly, though, it appears to me as if seti's current mapper is unable to resolve wildcards like `tsconfig.*.json`, isn't it? I'd be happy to see this feature being added.

**Comment by orta** [comment]
Yeah, it only supports name literals I think- This might be a constraint at the editor level rather than in here.

**Comment by Neonit** [comment]
Could be an editor constraint, but I know for sure that the `files.associations` setting in VS Code supports such patterns. I can assign a type to `tsconfig.*.json` there.

I would have fixed my problem using this, but unfortunately *tsconfig* is not registered as type in VS Code and it only supports registered type IDs there.

But unless VS Code uses different approaches to file matching in different locations it is likely to not be an editor constraint.

**Comment by orta** [comment]
@JacksonKearl do you know if this is feasible? ^

**Comment by JacksonKearl** [comment]
Unfortunately our file icon machinery currently only supports exact name match and extension match:
https://github.com/microsoft/vscode/blob/f91c7ded03bbf0bb3617bac4f4b9f9331add5b6f/src/vs/workbench/services/themes/browser/fileIconThemeData.ts#L299-L330

@aeschli might be more able to assist here.

**Comment by aeschli** [comment]
The issue for that is https://github.com/microsoft/vscode/issues/12493
